dnf/base.py

```diff
@@ -1744,9 +1753,111 @@ def install_module(self, specs, strict=True):
         """
         return self.repo_module_dict.install(specs, strict)
 
+    def classify(self, exclude, install):
+        install_specs = argparse.Namespace()
+        exclude_specs = argparse.Namespace()
+        classify_specs(install_specs, install)
+        classify_specs(exclude_specs, exclude)
+
+        return exclude_specs, install_specs
+
+    def exclude_spec(self, exclude_specs):
+        glob_excludes = [exclude for exclude in exclude_specs.pkg_specs
+                         if dnf.util.is_glob_pattern(exclude)]
+        excludes = [exclude for exclude in exclude_specs.pkg_specs
+                    if exclude not in glob_excludes]
+        exclude_query = self.sack.query().filter(name=excludes)
+        glob_exclude_query = self.sack.query().filter(name__glob=glob_excludes)
+        self.sack.add_excludes(exclude_query)
+        self.sack.add_excludes(glob_exclude_query)
+
+    def install_packages(self, package_specs, ignored):
+        for spec in package_specs:
+            try:
+                self.install(spec, strict=self.conf.strict)
+            except dnf.exceptions.Error:
+                ignored.append(spec)
+
+    def install_modules(self, module_specs):
+        # temporary hack, in future raise exception/add to ignored
+        # when enabling different stream for kickstart
+        former_assumeyes = self.conf.assumeyes
+        if not self.output:
+            self.conf.assumeyes = True
+        groups = self.install_module(module_specs)
+        self.conf.assumeyes = former_assumeyes
+        return groups
+
+    def exclude_groups(self, group_specs):
+        group_excludes = []
+
+        for group_spec in group_specs:
+            if '/' in group_spec:
+                split = group_spec.split('/')
+                group_spec = split[0]
+
+            if group_spec[0] == "^":
+                environment = self.comps.environment_by_pattern(group_spec[1:])
+                if not environment:
+                    continue
+
+            for group in environment.groups_iter():
+                for pkg in group.packages_iter():
+                    group_excludes.append(pkg.name)
+
+        exclude_query = self.sack.query().filter(name=group_excludes)
+        self.sack.add_excludes(exclude_query)
+
+    def install_groups(self, group_specs, exclude_packages, ignored):
+        for group_spec in group_specs:
+            try:
+                types = self.conf.group_package_types
+
+                if '/' in group_spec:
+                    split = group_spec.split('/')
+                    group_spec = split[0]
+                    types = split[1].split(',')
```

Comment:
Is this still a requirement from Anaconda team? I mean to provide group with types?

Reply:
Requirement from Anaconda.